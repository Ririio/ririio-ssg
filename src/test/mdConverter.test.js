const path = require('path')
const { processLine, mdConverter } = require('../text-converter/mdConverted')

describe('mark down converter', () => {
  describe('Line processor', () => {
    test('for heading', () => {
      expect(processLine('# Hello world!')).toMatch(/<h1>/)
    })

    test('for italic', () => {
      expect(processLine('_Hello world!_')).toMatch(/<i>/)
    })

    test('for bold', () => {
      expect(processLine('**Hello world!**')).toMatch(/<b>/)
    })

    test('for url', () => {
      expect(processLine('[url](https://google.com)')).toMatch(/[.*](.*)/)
    })
  })

  test('processed markdown match snapshot', () => {
    const testFileName = path.join(__dirname, '../../text-files/test.md')
    expect(mdConverter(testFileName)).toMatchSnapshot()
  })
})
