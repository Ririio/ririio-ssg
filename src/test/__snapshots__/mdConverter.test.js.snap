// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mark down converter processed markdown match snapshot 1`] = `
{
  "result": "<h1>Hello world!</h1>

<hr>

This text is <i>italic</i> and this is <b>bold</b>
the <a href="https://google.com">url</a> looks like that
",
  "storyTitle": "test.md",
}
`;
